docs/examples/self_attentive_sentence_embedding/self_attentive_sentence_embedding.md

```diff
@@ -0,0 +1,535 @@
+                l = l + penal_coeff * temp.norm(axis=(1, 2))
+
+            # backward calculate
+            l.backward()
+
+            # clip gradient
+            clip_params = [p.data() for p in model.collect_params().values()]
+            if clip is not None:
+                norm = nd.array([0.0], ctx)
+                for param in clip_params:
+                    if param.grad is not None:
+                        norm += (param.grad ** 2).sum()
+                norm = norm.sqrt().asscalar()
+                if norm > clip:
+                    for param in clip_params:
+                        if param.grad is not None:
+                            param.grad[:] *= clip / norm
+
+            # update parmas
+            trainer.step(batch_x.shape[0])
+
+            # keep result for metric
+            batch_pred = np.argmax(nd.softmax(batch_pred, axis=1).asnumpy(), axis=1)
+            batch_true = np.reshape(batch_y.asnumpy(), (-1, ))
+            total_pred.extend(batch_pred.tolist())
+            total_true.extend(batch_true.tolist())
+            batch_train_loss = l.mean().asscalar()
+
+            n_batch += 1
+            train_loss += batch_train_loss
+
+            if n_batch % 400 == 0:
+                print('epoch %d, batch %d, batch_train_loss %.4f, batch_train_acc %.3f' %
+                      (epoch, n_batch, batch_train_loss, accuracy_score(batch_true, batch_pred)))
+
+        # metric
+        F1_train = f1_score(np.array(total_true), np.array(total_pred), average='weighted')
+        acc_train = accuracy_score(np.array(total_true), np.array(total_pred))
+        train_loss /= n_batch
+
+        # evaluate on valid_data
+        F1_valid, acc_valid, valid_loss = evaluate(data_iter_valid, model, loss, ctx,
+                                                   penal_coeff, class_weight, loss_name)
+
+        print('epoch %d, learning_rate %.5f \n\t train_loss %.4f, acc_train %.3f, F1_train %.3f, ' %
+              (epoch, trainer.learning_rate, train_loss, acc_train, F1_train))
+        print('\t valid_loss %.4f, acc_valid %.3f, F1_valid %.3f, '
+              '\ntime %.1f sec' % (valid_loss, acc_valid, F1_valid, time.time() - start))
+        print('='*50)
+
+        # declay lr
+        if epoch % 2 == 0:
+            trainer.set_learning_rate(trainer.learning_rate * 0.9)
+```
+
+## Train
+Now that we are training the model, we use WeightedSoftmaxCE to alleviate the problem of data category imbalance. We performed statistical analysis on the data in advance to get a set of class_weight.
+
+```{.python .input}
+class_weight = None
+loss_name = 'wsce'
+optim = 'adam'
+lr = 0.001
+penal_coeff = 0.1
+clip = 0.5
+nepochs = 4
+
+trainer = gluon.Trainer(model.collect_params(), optim, {'learning_rate': lr})
+
+if loss_name == 'sce':
+    loss = gluon.loss.SoftmaxCrossEntropyLoss()
+elif loss_name == 'wsce':
+    loss = WeightedSoftmaxCE()
+    class_weight = nd.array([3.0, 5.3, 4.0, 2.0, 1.0], ctx=ctx)
```

Comment:
Where do these numbers come from?

Reply:
The value of class_weight is obtained by counting data in advance. It can be seen as a hyperparameter. Different data should have different class_weight.